src/seedu/addressbook/Main.java

```diff
@@ -84,10 +84,16 @@ private void runCommandLoopUntilExitCommand() {
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
 
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (StorageOperationException e) {
+                ui.showToUser(e.getMessage());
```

Comment:
As discussed in class, the control goes to showToUser method that prints a friendly error message. Is there something else we can do by handling exceptions more gracefully? Maybe save the program state to give the user a chance to change the file permission?